libraries/hotstuff/include/eosio/hotstuff/qc_chain.hpp

```diff
@@ -30,6 +33,34 @@ namespace eosio::hotstuff {
    using namespace boost::multi_index;
    using namespace eosio::chain;
 
+   template <typename StateObj>
+   static void read_state(const std::string file_path, StateObj& s){
+
+      if (file_path != std::string()){
+         fc::cfile pfile;
+         pfile.set_file_path(file_path);
+         pfile.open("rb");
+         auto ds = pfile.create_datastream();
+         fc::raw::unpack(ds, s);
+         pfile.close();
+      }
+      
+   }
+
+   template <typename StateObj>
+   static void write_state(const std::string file_path, const StateObj s){
+
+      if (file_path != std::string()){
+        fc::cfile pfile;
+        pfile.set_file_path(file_path);
+        pfile.open(fc::cfile::truncate_rw_mode);
+        auto data = fc::raw::pack(s);
+        pfile.write(data.data(), data.size());
+        pfile.close();
```

Comment:
I don't think we want to `open`/`close` the file for each write. Better to just `flush()` after each write.